File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/Appointment.ts
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type AppointmentParticipant = BackboneElement & {
    type?: CodeableConcept[];
    actor?: Reference;
    period?: Period;
    status: string;
    required?: string;
};

export type Appointment = DomainResource & {
    description?: string;
    serviceCategory?: CodeableConcept[];
    slot?: Reference[];
    specialty?: CodeableConcept[];
    cancelationReason?: CodeableConcept;
    requestedPeriod?: Period[];
    patientInstruction?: string;
    start?: string;
    reasonCode?: CodeableConcept[];
    created?: string;
    participant: AppointmentParticipant[];
    serviceType?: CodeableConcept[];
    supportingInformation?: Reference[];
    priority?: number;
    appointmentType?: CodeableConcept;
    status: string;
    comment?: string;
    minutesDuration?: number;
    identifier?: Identifier[];
    basedOn?: Reference[];
    end?: string;
    reasonReference?: Reference[];
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Appointment.py
from typing import Optional, List
from pydantic import *
from ..base.Period import Period
from ..base.CodeableConcept import CodeableConcept
from ..base.DomainResource import DomainResource
from ..base.Reference import Reference
from ..base.Identifier import Identifier
from ..base.BackboneElement import BackboneElement

class Appointment_Participant(BackboneElement):
    type: Optional[List[CodeableConcept]] = None
    actor: Optional[Reference] = None
    period: Optional[Period] = None
    status: str
    required: Optional[str] = None

class Appointment(DomainResource):
    description: Optional[str] = None
    service_category: Optional[List[CodeableConcept]] = None
    slot: Optional[List[Reference]] = None
    specialty: Optional[List[CodeableConcept]] = None
    cancelation_reason: Optional[CodeableConcept] = None
    requested_period: Optional[List[Period]] = None
    patient_instruction: Optional[str] = None
    start: Optional[str] = None
    reason_code: Optional[List[CodeableConcept]] = None
    created: Optional[str] = None
    participant: list[Appointment_Participant] = []
    service_type: Optional[List[CodeableConcept]] = None
    supporting_information: Optional[List[Reference]] = None
    priority: Optional[int] = None
    appointment_type: Optional[CodeableConcept] = None
    status: str
    comment: Optional[str] = None
    minutes_duration: Optional[int] = None
    identifier: Optional[List[Identifier]] = None
    based_on: Optional[List[Reference]] = None
    end: Optional[str] = None
    reason_reference: Optional[List[Reference]] = None